src/fastfield/mod.rs

```diff
@@ -1068,13 +1068,94 @@ mod tests {
         assert!(column_without_opt.is_none());
-        let column_with_opt: Option<StrColumn> = fast_fields.str("with\u{1}hello").unwrap();
+        let column_with_opt: Option<StrColumn> = fast_fields.str("with.hello").unwrap();
         let column_with: StrColumn = column_with_opt.unwrap();
         assert!(column_with.term_ords(0).next().is_none());
         assert!(column_with.term_ords(1).eq([0]));
         assert!(column_with.term_ords(2).eq([2]));
         assert!(column_with.term_ords(3).eq([1]));
     }
+
+    #[test]
+    fn test_fast_field_in_json_field_no_expand_dots_disabled() {
+        let mut schema_builder = Schema::builder();
+        let json_option = JsonObjectOptions::default().set_fast();
+        let json = schema_builder.add_json_field("json", json_option);
+        let schema = schema_builder.build();
+        let index = Index::create_in_ram(schema);
+        let mut index_writer = index.writer_for_tests().unwrap();
+        index_writer
+            .add_document(doc!(json => json!({"attr.age": 32})))
+            .unwrap();
+        index_writer.commit().unwrap();
+        let reader = index.reader().unwrap();
+        let searcher = reader.searcher();
+        let reader = searcher.segment_reader(0u32);
+        let fast_field_reader = reader.fast_fields();
+        assert!(fast_field_reader
+            .column_opt::<i64>("json.attr.age")
+            .unwrap()
+            .is_none());
+        let column = fast_field_reader
+            .column_opt::<i64>(r#"json.attr\.age"#)
+            .unwrap()
+            .unwrap();
+        let vals: Vec<i64> = column.values_for_doc(0u32).collect();
+        assert_eq!(&vals, &[32])
+    }
+
+    #[test]
+    fn test_fast_field_in_json_field_expand_dots_enabled() {
+        let mut schema_builder = Schema::builder();
+        let json_option = JsonObjectOptions::default()
+            .set_fast()
+            .set_expand_dots_enabled();
+        let json = schema_builder.add_json_field("json", json_option);
+        let schema = schema_builder.build();
+        let index = Index::create_in_ram(schema);
+        let mut index_writer = index.writer_for_tests().unwrap();
+        index_writer
+            .add_document(doc!(json => json!({"attr.age": 32})))
+            .unwrap();
+        index_writer.commit().unwrap();
+        let reader = index.reader().unwrap();
+        let searcher = reader.searcher();
+        let reader = searcher.segment_reader(0u32);
+        let fast_field_reader = reader.fast_fields();
+        for test_column_name in &["json.attr.age", "json.attr\\.age"] {
+            let column = fast_field_reader
+                .column_opt::<i64>(test_column_name)
+                .unwrap()
+                .unwrap();
+            let vals: Vec<i64> = column.values_for_doc(0u32).collect();
+            assert_eq!(&vals, &[32]);
+        }
+    }
+
+    #[test]
+    fn test_fast_field_dot_in_schema_field_name() {
```

Comment:
can we add a test for json field with dot

Reply:
added a unit test